torch_geometric/nn/models/explainer.py

```diff
@@ -114,8 +114,165 @@ def forward(self, mask, *args):
+    return x_attr_dict, edge_attr_dict
+
+
+# TODO(jinu) Is there any point of inheriting from `CaptumModel`
+class CaptumHeteroModel(CaptumModel):
+    def __init__(self, model: torch.nn.Module, mask_type: str, output_id: int,
+                 metadata: Metadata):
+        super().__init__(model, mask_type, output_id)
+        self.node_types = metadata[0]
+        self.edge_types = metadata[1]
+
+    def _captum_data_to_hetero_data(
+        self, *args
+    ) -> Tuple[Dict[NodeType, Tensor], Dict[EdgeType, Tensor], Optional[Dict[
+            EdgeType, Tensor]]]:
+        """Converts tuple of tensors to `x_dict`, `edge_index_dict` and
+        `edge_mask_dict`."""
+        num_edge_types = len(self.edge_types)
+        num_node_types = len(self.node_types)
+
+        if self.mask_type == 'node':
+            node_tensors = args[0:num_node_types]
+            node_tensors = [mask.squeeze(0) for mask in node_tensors]
+            x_dict = dict(zip(self.node_types, node_tensors))
+            edge_index_dict = args[num_node_types]
+        elif self.mask_type == 'edge':
+            edge_mask_tensors = args[0:num_edge_types]
+            x_dict = args[num_edge_types]
+            edge_index_dict = args[num_edge_types + 1]
+        else:
+            node_tensors = args[0:num_node_types]
+            node_tensors = [mask.squeeze(0) for mask in node_tensors]
+            x_dict = dict(zip(self.node_types, node_tensors))
+            edge_mask_tensors = args[num_node_types:num_node_types +
+                                     num_edge_types]
+            edge_index_dict = args[num_node_types + num_edge_types]
+
+        if 'edge' in self.mask_type:
+            edge_mask_tensors = [mask.squeeze(0) for mask in edge_mask_tensors]
+            edge_mask_dict = dict(zip(self.edge_types, edge_mask_tensors))
+        else:
+            edge_mask_dict = None
+        return x_dict, edge_index_dict, edge_mask_dict
+
+    def forward(self, *args):
+        # Validate args:
+        num_nodes = len(self.node_types)
+        num_edges = len(self.edge_types)
+        if self.mask_type == "node":
+            assert len(args) >= num_nodes + 1
+            len_remaining_args = len(args) - (num_nodes + 1)
+        elif self.mask_type == "edge":
+            assert len(args) >= num_edges + 2
+            len_remaining_args = len(args) - (num_edges + 2)
+        else:
+            assert len(args) >= num_nodes + num_edges + 1
+            len_remaining_args = len(args) - (num_edges + num_edges + 1)
+
+        # Get main args:
+        (x_dict, edge_index_dict,
+         edge_mask_dict) = self._captum_data_to_hetero_data(*args)
+
+        if 'edge' in self.mask_type:
+            set_hetero_masks(self.model, edge_mask_dict, edge_index_dict)
+
+        if len_remaining_args > 0:
+            # If there are args other than `x_dict` and `edge_index_dict`
+            x = self.model(x_dict, edge_index_dict, *args[-len_remaining_args])
```

Comment:
```suggestion
            x = self.model(x_dict, edge_index_dict, *args[-len_remaining_args:])
```
Can't we call the model using the args even if they are empty? So regardless of whether len_remaining_args > 0 or not? And thus combine the if/else to one line.